core/root/stack.go

```diff
@@ -48,10 +40,24 @@ func (r stackRendererImpl) RenderFiles() error {
 		filegen.File(defaults.EtcdStackTemplateTmplFile, etcd.StackTemplateTemplate, 0644),
 		filegen.File(defaults.NodePoolStackTemplateTmplFile, nodepool.StackTemplateTemplate, 0644),
 		filegen.File(defaults.RootStackTemplateTmplFile, config.StackTemplateTemplate, 0644),
-		filegen.File("kubeconfig", kubeconfig.Bytes(), 0600),
+		filegen.File("kubeconfig", kubeconfig, 0600),
 	); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func parseKubeconfig(clusterConfig *controlplane.Config) ([]byte, error) {
```

Comment:
Not in this PR but I'd prefer naming it `generateKubeconfig`, because what this parses is actually kubeconfig template, rather than kubeconfig itself

Reply:
`generateKubeconfigTmpl` or `parseKubeconfigTmpl`, or just `generateKubeconfig`, lets rename it otherwise it will get forgotten

Reply:
Thanks for the response!

Just sharing my sentiment:

> generateKubeconfigTmpl

sounds like it is generating a template, which isn't true

> parseKubeconfigTmpl

it parses the template in order to generate a kubeconfig. so, I'd rather include the goal of the function(=generate kubeconfig) into its name